wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt

```diff
@@ -77,12 +78,25 @@ sealed class Target : Serializable {
       protoFile: ProtoFile,
       emittingRules: EmittingRules,
       claimedDefinitions: ClaimedDefinitions,
+      claimedPaths: MutableMap<Path, Type>,
       isExclusive: Boolean
     ) {
       protoFile.types
           .filter { it !in claimedDefinitions && emittingRules.includes(it.type) }
           .forEach { type ->
-            handle(type)
+            val generatedFilePath = handle(type)
+
+            if (generatedFilePath != null) {
+              if (generatedFilePath in claimedPaths.keys) {
+                val other = claimedPaths[generatedFilePath]!!
```

Comment:
You shouldn't call `contains` and then `get` on a map. Just call `get` and then check if the result is non-null. Otherwise you pay the hashing and traversal cost twice. Here you're actually paying it twice in both branches because the other codepath then inserts an entry. This can all be consolidated into a single interaction with the map:
```kotlin
claimedPaths.putIfAbsent(generatedFilePath, type)?.let { other ->
  throw ISE(...)
}
```

Reply:
fixed